pypeman/channels.py

```diff
@@ -656,22 +661,37 @@ async def watch_for_file(self):
                         status = self.file_status(filename)
 
                         if status in [FileWatcherChannel.MODIFIED, FileWatcherChannel.NEW]:
-                            filepath = os.path.join(self.basedir, filename)
-                            self.data[filename] = os.stat(filepath).st_mtime
+                            filepath = pathlib.Path(self.basedir) / filename
+                            self.data[filename] = filepath.stat().st_mtime
+                            if self.real_extensions:
+                                for extension in self.real_extensions:
+                                    real_fpath = filepath.with_suffix(extension)
+                                    if real_fpath.exists():
+                                        filepath = real_fpath
+                                        filename = real_fpath.name
+                                        break
+                                else:
+                                    # If no related files
+                                    # TODO : ask if raise exc or not
+                                    logger.error(
+                                        "No %r related file to %s",
+                                        self.real_extensions, str(filepath))
+                                    continue
 
                             # Read file and make message
                             if self.binary_file:
                                 mode = "rb"
                             else:
                                 mode = "r"
 
-                            with open(filepath, mode) as f:
-                                msg = message.Message()
+                            msg = message.Message()
+
+                            with open(str(filepath), mode) as f:
```

Comment:
open accepts path objects, so you can do directly:

```
with open(filepath, mode) as f:
```

but you can also do 

```
msg.payload = filepath.read_bytes() if binary_mode else filepath.read()
```

Reply:
done